core/src/main/java/org/eclipse/hono/util/TenantConstants.java

```diff
@@ -74,13 +74,13 @@ public static JsonObject getTenantMsg(final Message message) {
     /**
      * Gets a JSON object representing the reply to a Tenant API request via the vert.x event bus.
      *
-     * @param tenantId The tenant for which the message was processed. Must not be null.
+     * @param tenantId The tenant for which the message was processed. If null, the value for the key
+     *                 {@link RequestResponseApiConstants#FIELD_TENANT_ID} in the reply will be null, too.
```

Comment:
???
why would we want to do that? according to the spec the *tenant-id* is mandatory (which FMPOV implies that is must not be `null`).

Reply:
If we return a 400 there is no need to supply a payload that conforms to the 200 case. We should thus either return an empty payload or a payload containing a detail message explaining the problem

Reply:
Agree to that.
Since the other APIs (registration and credentials) are affected by this behaviour as well, I would keep that outside this PR and create a solution later for all APIs in a small other PR.